### PR TITLE
[AAP-8240] Tests, and Rbac for Templates and TemplatesList.

### PR DESCRIPTION
**Pull request by daphnemaeve**
Jira ticket: https://issues.redhat.com/browse/AAP-8240

Adds tests for TemplatesList component in TemplateList.cy.tsx.

Also adds rbac for Templates and TemplatesList to disable Create Template button when the user doesn't have permission to do so.

### CODE REVIEW

File: frontend/awx/resources/templates/TemplatesList.cy.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+describe('TemplatesList', () => {
+  describe('ErrorList', () => {
+    it('displays error if templates list is not successfully loaded', () => {
+      cy.intercept({ method: 'GET', url: '/api/v2/projects/6/*' }, { statusCode: 500 });
+      cy.mount(<TemplatesList url={'/api/v2/projects/6/'} />);
+      cy.contains('Error loading templates');
+    });
+  });
+
+  describe('Non-empty list', () => {
```
Comment (vishsanghishetty) [inline frontend/awx/resources/templates/TemplatesList.cy.tsx:14]: @daphnemaeve is there a test that covers empty state error message when there are no templates to display?

**Review by AlexSCorey — CHANGES_REQUESTED** [review]
This is a really good start.  I left some comments below.  I am happy to clarify any, or help in any way.  Just ping me.

File: frontend/awx/resources/templates/TemplatesList.cy.tsx (hunk 1 above)
Comment (daphnemaeve) [reply frontend/awx/resources/templates/TemplatesList.cy.tsx:14]: There is a test to display an error if the list is not correctly loaded, but I can add one to test the behavior if the list loads but has no templates

File: frontend/awx/resources/templates/TemplatesList.cy.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+                filterable: true,
+              },
+            },
+          },
+        },
+      }));
+      cy.mount(<TemplatesList url={'/api/v2/projects/6/*'} />);
+      cy.contains('.pf-v5-c-dropdown__toggle', 'Create template').should('not.be.disabled');
+    });
+
```
[the remark this answers is not in the document]
Comment (AlexSCorey) [reply frontend/awx/resources/templates/TemplatesList.cy.tsx:115]: @daphnemaeve and I met, and discussed this and we are good to go.

File: frontend/awx/resources/templates/TemplatesList.cy.tsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    it('Launch action item should call API /launch endpoint', () => {
+      cy.intercept(
+        { method: 'GET', url: '/api/v2/job_templates/7/launch/' },
+        { fixture: 'jobTemplateLaunch' }
+      ).as('launchRequest');
+      cy.mount(<TemplatesList url={'/api/v2/projects/6/*'} />);
+      cy.clickTableRowPinnedAction('Demo Job Template', 'launch-template');
+      cy.wait('@launchRequest');
+    });
```
[the remark this answers is not in the document]
Comment (AlexSCorey) [reply frontend/awx/resources/templates/TemplatesList.cy.tsx]: Please ignore this recommendation.  You can keep the test as it, or remove it.  We have it covered in the e2e tests.

File: frontend/awx/resources/templates/TemplatesList.cy.tsx (hunk 1 above)
Comment (daphnemaeve) [reply frontend/awx/resources/templates/TemplatesList.cy.tsx]: @vishsanghishetty Added tests for empty list! Please let me know if they look good to you

**Review by AlexSCorey — APPROVED** [review]
Just remove the jobTemplateLaunchPOST.json file, and looks like you are good to go!.

**Pull request merged**